From a2c230ac9f33339a3175e614c788a89407b2e5fa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ji=C5=99=C3=AD=20Klime=C5=A1?= <jklimes@redhat.com>
Date: Thu, 19 Feb 2015 08:46:24 +0100
Subject: [PATCH] device: remove incorrect assertions (rh #1181207)
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We would crash in the following case, for example:
* configure bond0 with slaves eth0 and eth0
* up all interfaces
* rm /etc/sysconfig/network-scripts/ifcfg-bond0
* rm /etc/sysconfig/network-scripts/ifcfg-eth0
(* rm /etc/sysconfig/network-scripts/ifcfg-eth1)
* BOOM

The assertions were removed upstream by these commits:
f93c15416152c1d071102cce716d82a1d3616c44
08e0cfb484dfc6aa6342871158ffe752c7c50f03

https://bugzilla.redhat.com/show_bug.cgi?id=1181207
Signed-off-by: Jiří Klimeš <jklimes@redhat.com>
---
 src/nm-device.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/nm-device.c b/src/nm-device.c
index 02a6036..d68192b 100644
--- a/src/nm-device.c
+++ b/src/nm-device.c
@@ -657,9 +657,6 @@ slave_state_changed (NMDevice *slave,
 	            slave_new_state,
 	            state_to_string (slave_new_state));
 
-	g_assert (priv->state > NM_DEVICE_STATE_DISCONNECTED);
-	g_assert (priv->state <= NM_DEVICE_STATE_ACTIVATED);
-
 	/* Don't try to enslave slaves until the master is ready */
 	if (priv->state < NM_DEVICE_STATE_CONFIG)
 		return;
-- 
2.1.0

